include/exadg/fluid_structure_interaction_precice/solver.h

```diff
@@ -0,0 +1,208 @@
+                        Patterns::Integer(1,EXADG_DEGREE_MAX),
+                        true);
+      prm.add_parameter("RefineFluid",
+                        refine_fluid,
+                        "Number of mesh refinements (fluid).",
+                        Patterns::Integer(0,20),
+                        true);
+      prm.add_parameter("RefineStructure",
+                        refine_structure,
+                        "Number of mesh refinements (structure).",
+                        Patterns::Integer(0,20),
+                        true);
+    prm.leave_subsection();
+    // clang-format on
+  }
+
+  unsigned int degree_fluid = 3, degree_structure = 3;
+
+  unsigned int refine_fluid = 0, refine_structure = 0;
+};
+
+
+void
+create_input_file(std::string const & input_file)
+{
+  dealii::ParameterHandler prm;
+
+  GeneralParameters general;
+  general.add_parameters(prm);
+
+  ResolutionParameters resolution;
+  resolution.add_parameters(prm);
+
+  Parameters::PreciceAdapterConfiguration precice_parameters;
+  precice_parameters.add_parameters(prm);
+
+  // we have to assume a default dimension and default Number type
+  // for the automatic generation of a default input file
+  constexpr int Dim = 2;
+  using Number      = double;
+
+  try
+  {
+    FSI::get_application<Dim, Number>(input_file, MPI_COMM_WORLD)->add_parameters(prm);
+  }
+  catch(...)
+  {
+  }
+
+  prm.print_parameters(input_file,
+                       dealii::ParameterHandler::Short |
+                         dealii::ParameterHandler::KeepDeclarationOrder);
+}
+
+
+template<int dim, typename Number>
+void
+run(std::string const & input_file, MPI_Comm const & mpi_comm, bool const is_test)
+{
+  Timer timer;
+  timer.restart();
+
+  std::shared_ptr<FSI::ApplicationBase<dim, Number>> application =
+    FSI::get_application<dim, Number>(input_file, mpi_comm);
+
+
+  ExaDG::ResolutionParameters resolution(input_file);
+  application->set_parameters_convergence_study(resolution.degree_fluid,
+                                                resolution.degree_structure,
+                                                resolution.refine_fluid,
+                                                resolution.refine_structure);
+
+  Parameters::PreciceAdapterConfiguration precice_param;
+  dealii::ParameterHandler                prm;
+  precice_param.add_parameters(prm);
+  prm.parse_input(input_file, "", true, true);
+
+  bool const is_solid = precice_param.physics == "Structure";
+
+  std::shared_ptr<FSI::Driver<dim, Number>> driver;
+
+  if(is_solid)
+    driver =
+      std::make_shared<FSI::DriverSolid<dim, Number>>(input_file, mpi_comm, application, is_test);
+  else
+    driver =
+      std::make_shared<FSI::DriverFluid<dim, Number>>(input_file, mpi_comm, application, is_test);
+
+  driver->setup();
+
+  driver->solve();
+
+  if(!is_test)
+    driver->print_performance_results(timer.wall_time());
+}
+} // namespace ExaDG
+
+int
+main(int argc, char ** argv)
```

Comment:
how is the precice-FSI executable called? I only see one input file here, but two files "fluid-exadg/input.json" and "solid-exadg/input.json"